src/thermostats/TemperatureObserver.h

```diff
@@ -0,0 +1,142 @@
+//
+// Created by alex on 06.03.24.
+//
+
+#ifndef MARDYN_TEMPERATUREOBSERVER_H
+#define MARDYN_TEMPERATUREOBSERVER_H
+
+
+#include "utils/xmlfileUnits.h"
+#include "utils/mardyn_assert.h"
+
+#include <array>
+#include <vector>
+
+
+class Domain;
+class ParticleContainer;
+class Ensemble;
+
+/**
+ * Ring buffer that stores up to n entries. Will override oldest entry once n+1 elements are inserted.
+ * */
+template<typename T>
+struct NBuffer {
```

Comment:
Did not check 100% the usage of this but maybe have a look at [utils/Accumulator.h](https://github.com/ls1mardyn/ls1-mardyn/blob/master/src/utils/Accumulator.h) to see if there is already functionality available.

Reply:
I tried using it but failed to compile afterwards due to redefinition in TemperatureControl.
Moving include directives to cpp files and using predeclarations did not help.
So, for now I'll keep it as is.